### ISSUE TITLE
cmd/govim: incremental buffer updates fail when trying to format-on-save

### ISSUE THREAD

**Issue by myitcv** [myitcv/govim#251]
With thanks to @leitzler for the report. As a result of this bug I quickly submitted #250 to invert the default and disable incremental updates for now. More tests definitely needed.

```
$ go mod init mod.com
$ cat > main.go <<EOD
package main

func main() {
    asd
}
EOD
```

Then open the file in Vim (following #250):

```
$ GOVIM_DISABLE_INCREMENTALSYNC=false vi main.go
```

and simply `:w`.

Vim locks up, with the following error message in the logs:

```
Caught 'Vim(let):E523: Not allowed here' in function <SNR>46_callbackAutoCommand[6]..GOVIM_internal_EnrichDelta[14]..GOVIM_internal_BufChanged[2]..<SNR>46_callbackFunction[2]..GOVIM_internal_EnrichDelta[14]..GOVIM_internal_BufChanged[2]..<SNR>46_callbackFunction[2]..<SNR>46_define, line 54
```

**Comment by myitcv** [comment]
Raised https://github.com/vim/vim/issues/4438

**Comment by myitcv** [comment]
This was fixed in #256
